synapseclient/models/activity.py

```diff
@@ -1 +1,345 @@
-# TODO
+import asyncio
+from dataclasses import dataclass, field
+from typing import List, Dict, Union, Tuple
+
+from synapseclient.activity import Activity as Synapse_Activity
+from opentelemetry import trace, context
+
+from typing import Optional, TYPE_CHECKING
+
+from synapseclient import Synapse
+from synapseclient.core.async_utils import otel_trace_method
+
+if TYPE_CHECKING:
+    from synapseclient.models import File, Table
+
+USED_URL_CONCRETE_TYPE = "org.sagebionetworks.repo.model.provenance.UsedURL"
+USED_ENTITY_CONCRETE_TYPE = "org.sagebionetworks.repo.model.provenance.UsedEntity"
+
+tracer = trace.get_tracer("synapseclient")
+
+
+@dataclass
+class UsedEntity:
+    """
+    Reference to a Synapse entity that was used or executed by an Activity.
+
+    Attributes:
+        target_id: The ID of the entity to which this reference refers.
+        target_version_number: The version number of the entity to which this reference refers.
+    """
+
+    target_id: Optional[str] = None
+    """The the id of the entity to which this reference refers."""
+
+    target_version_number: Optional[int] = None
+    """The version number of the entity to which this reference refers."""
+
+
+@dataclass
+class UsedURL:
```

Comment:
Nit: UsedURL is: https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/provenance/UsedURL.html

Reply:
Discussed in other comments on why this is the way it is.